packages/angular/ssr/src/app.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import { Hooks } from './hooks';
+import { AngularAppManifest } from './manifest';
+import { render } from './render';
+
+/**
+ * Configuration options for initializing a `AngularServerApp` instance.
+ */
+export interface AngularServerAppOptions {
+  /**
+   * Indicates whether the application is in development mode.
+   *
+   * When set to `true`, the application runs in development mode with additional debugging features.
+   */
+  isDevMode?: boolean;
+
+  /**
+   * Optional hooks for customizing the server application's behavior.
+   */
+  hooks?: Hooks;
```

Comment:
Quick question: would it be a part of the public API? If yes, I'm curious if this is needed to cover some existing use-cases that developers have with the current APIs? Just wanted to see if we can reduce the public API and add this feature later, based on the feedback.

Reply:
Primarily, hooks are needed during `ng serve` to transform `index.html` using the Vite API to add HMR/live-reload scripts.

Additionally, in production environments, these hooks can be used to replace or inject `nonce` values.

Reply:
The mentioned scenarios look like implementation-specific details. If there are no known app use-cases that we won't be able to support without exposing hooks into the public API - I'd vote for **not** including them into the public API for now.